### ISSUE TITLE
TestForm throws InvalidHeaderValue with any word with letter r in filename or name fields

### ISSUE THREAD

**Issue by makorne** [poem-web/poem#416]
## Expected Behavior
No errors

## Actual Behavior
poem/src/test/form.rs:157] &content_disposition = "form-data; name=\"\\\r\""

called `Result::unwrap()` on an `Err` value: InvalidHeaderValue
thread 'test::form::tests::multipart' panicked at 'called `Result::unwrap()` on an `Err` value: InvalidHeaderValue', poem/src/test/form.rs:160:45

## Steps to Reproduce the Problem

    #[tokio::test]
    async fn multipart() {
        let mut a = Vec::new();
        TestForm::new()
            .field(TestFormField::text("1").filename("r"))
            .field(TestFormField::text("2").name("r"))
            .into_async_read()
            .read_to_end(&mut a)
            .await
            .unwrap();

```
fn legal_str(s: impl AsRef<str>) -> String {
    s.as_ref()
        .replace('\\', "\\\\")
        .replace('\"', "\\\"")
        .replace('r', "\\\r")        <<<<< The error occurs because of this wrong replacing. Looks like it should be  .replace('\r' ...  
        .replace('\n', "\\\n")
}
```

**Comment by sunli829** [comment]
Thanks, fixed in `master` branch 😂

**Comment by sunli829** [comment]
Released in `v1.3.48`
